tutorials/upload-your-first-file-folder-with-ethfs-cli.md

```diff
@@ -0,0 +1,95 @@
+# Upload Your First File/Folder with ethfs-cli
+
+## **Introduction**
+
+In this tutorial, we will demonstrate how to upload files or folders using the **ethfs-cli** tool. We assume that
+
+* there is a folder to be uploaded; (dist)
+* there are two files in the folder. (hello.txt and img/1.jpeg)
+
+<figure><img src="../.gitbook/assets/image (17).png" alt=""><figcaption></figcaption></figure>
+
+## Step 1: Install ethfs-cli
+
+You can install ethfs-cli by the following command
+
+`npm i -g ethfs-cli`
+
+The npm page of ethfs-cli can be found [here](https://www.npmjs.com/package/ethfs-cli).
+
+## Step 2: Create the FlatDirectory Contract
+
+We want to create a [FlatDirectory](https://docs.web3url.io/advanced-topics/flatdirectory) with the private key 0x112233... on the QuarkChain L2 Testnet
+
+```
+ethfs-cli create -p 0x112233... -c 43069
+```
+
+If we want to create it on the other chains that support EVM, we need to change the chainId, such as the Sepolia.
+
+```
+ethfs-cli create -p 0x112233... -c 11155111
+```
+
+The RPC URL of the chain can also be specified.
+
+```
+ethfs-cli create -p 0x112233... -r https://...
+```
+
+You will get a FlatDirectory address: [<mark style="color:blue;">0x336fBd105C07DB2317149bAE143e378Ccdf56b3d</mark>](https://explorer.testnet.l2.quarkchain.io/address/0x336fBd105C07DB2317149bAE143e378Ccdf56b3d/transactions) after the transaction is confirmed.
+
+<figure><img src="../.gitbook/assets/img10.png" alt=""><figcaption></figcaption></figure>
+
+## Step 3: Upload Files
+
+In this section, you will upload the folder into the FlatDirectory that you just created.
+
+Run the command to upload the file.
+
+```
+ethfs-cli upload -a <address> -p <private-key> -f <directory|file> -c [chain-id] -t <upload-type>
+```
+
+The command to upload the contents of the "_dist"_ folder to the address "_0x336f...56b3d"_ is:
+
+```
+ethfs-cli upload -a 0x336f... -p 0x112233... -f /Users/.../dist -c 43069 -t 2
+```
+
+The execution results are as follows.
+
+<figure><img src="../.gitbook/assets/img9.png" alt=""><figcaption></figcaption></figure>
```

Comment:
The img9.png still show part of the private key. You can use $private_key instead